src/sources/aws_s3/mod.rs

```diff
@@ -145,16 +147,23 @@ enum CreateSqsIngestorError {
     RegionParse { source: rusoto::region::ParseError },
 }
 
-fn s3_object_decoder(
+/// None if body is empty
+async fn s3_object_decoder(
     compression: Compression,
     key: &str,
     content_encoding: Option<&str>,
     content_type: Option<&str>,
-    body: rusoto_s3::StreamingBody,
+    mut body: rusoto_s3::StreamingBody,
 ) -> Box<dyn tokio::io::AsyncRead + Send + Unpin> {
-    use async_compression::tokio::bufread;
+    let first = if let Some(first) = body.next().await {
+        first
+    } else {
+        return Box::new(tokio::io::empty());
+    };
```

Comment:
Could this be accomplished more easily with `peekable`?
```suggestion
    let mut body = body.peekable();
    if body.peek().is_none() {
        return Box::new(tokio::io::empty());
    };
```
Then there's no need to chain this `first` with the rest of the body.

Reply:
> Could this be accomplished more easily with peekable?

Yes, but the compiler was complaining that `body.peekable().into_async_read()` doesn't implement `AsyncRead`, lost some 2h on that at which point I just gave up and implemented it like this. Trimming 1-2 lines of that isn't worth the time, although it would be good to know what was the issue.